### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by felixtheflex**
## Jira ticket URL

- Just add the ticket number to the end:

https://dfedigital.atlassian.net/browse/TEVA-2762

## Changes in this PR:

Update documentation to reflect changes in the way certain aspects of our procedures

### CODE REVIEW

File: documentation/aws-roles-and-cli-tools.md [hunk 1]
```diff
@@ -310,3 +310,18 @@ export AWS_ACCESS_KEY_ID="$(echo $sessionInfo | jq '.Credentials.AccessKeyId' |
 export AWS_SECRET_ACCESS_KEY="$(echo $sessionInfo | jq '.Credentials.SecretAccessKey' | tr -d '"')"
 export AWS_SESSION_TOKEN="$(echo $sessionInfo | jq '.Credentials.SessionToken' | tr -d '"')"
 ```
+
+
+### Rotating `deploy` user access key
+
+
+The `deploy` user is an AWS account and it is used to run the CI\CD. Both its `access key` and `access_key_id` are stored on Github Secrets (AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY). For security reasons, the access key needs to be rotated occasionally.
```
Comment (david-mears-dfe) [inline documentation/aws-roles-and-cli-tools.md:318]: Any idea of how often this should be?

File: documentation/aws-roles-and-cli-tools.md [hunk 2]
```diff
@@ -310,3 +310,18 @@ export AWS_ACCESS_KEY_ID="$(echo $sessionInfo | jq '.Credentials.AccessKeyId' |
...
+
+The `deploy` user and its access key (` aws_iam_access_key` ) are deployed via the `common` terraform module. Please note, these resources are not deployed via the CI\CD pipeline.
```
Comment (david-mears-dfe) [inline documentation/aws-roles-and-cli-tools.md:320]: Both instances of `CI\CD` should be `CI/CD`

File: documentation/aws-roles-and-cli-tools.md [hunk 3]
```diff
@@ -310,3 +310,18 @@ export AWS_ACCESS_KEY_ID="$(echo $sessionInfo | jq '.Credentials.AccessKeyId' |
...
+
+To rotate the key, please do the following:-
+- [] From the root of the project, change directory - `cd terraform/common/`
+- [] `aws-vault exec Administrator -- terraform apply -replace aws_iam_access_key.deploy`
+- [] `aws-vault exec Administrator -- terraform output -json` - note the newly generated ACCESS_KEY_ID and ACCESS_KEY.
+- [] Copy the newly generated ACCESS_KEY_ID and ACCESS_KEY to Github secrets -  AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
+- [] Test by running\triggering a deploy workflow or run a workflow_dispatch action
```
Comment (david-mears-dfe) [inline documentation/aws-roles-and-cli-tools.md:327]: `/`

File: documentation/deployments.md [hunk 4]
```diff
@@ -8,14 +8,16 @@ This builds a Docker image from a feature branch, and then uses Terraform to cre
 
 - Push to a feature branch.
 - Create a Pull Request.
+- attach a `deploy` label
+
+The GitHub actions workflow [review.yml](/.github/workflows/review.yml) performs these steps, if a `deploy` label is attached to the pull request, then below happens, else no `build` or `deploy` jobs:
```
Comment (david-mears-dfe) [inline documentation/deployments.md:13]:
```suggestion
The GitHub actions workflow [review.yml](/.github/workflows/review.yml) performs the below steps if a `deploy` label is attached to the pull request. If not, the `build` and `deploy` jobs will not run:
```

File: documentation/deployments.md [hunk 5]
```diff
@@ -8,14 +8,16 @@ This builds a Docker image from a feature branch, and then uses Terraform to cre
...
 - Builds a Docker image from code in the feature branch
 - Tags the Docker image with the tags:
     - branch name (e.g. [TEVA-1797-update-school-type-summary](https://hub.docker.com/layers/dfedigital/teaching-vacancies/TEVA-1797-update-school-type-summary/images/sha256-dc01451b1486e40a3fb1a32ca577c65ece1a28a2ff27eefbd2455202c93caa71?context=explore)). Review apps often go through several iterations, so it's worth the creation of an image in order to speed up building from cache on subsequent pushes to the branch.
     - a composite tag containing the PR number and a unique timestamp (e.g. [review-pr-2664-80dd5f417a0faabfbe3f1a4bf8570eefec07139a-20210118172840](https://hub.docker.com/layers/dfedigital/teaching-vacancies/review-pr-2664-80dd5f417a0faabfbe3f1a4bf8570eefec07139a-20210118172840/images/sha256-dc01451b1486e40a3fb1a32ca577c65ece1a28a2ff27eefbd2455202c93caa71?context=explore)). This is useful for identifying the image.
-- Logs in to Docker Hub as the service account `teachingjobs`
-- Pushes the image to Docker Hub
+- Logs in to Github's container registy as the service account `twd-tv-ci`
+- Pushes the image to Github's container registy
```
Comment (david-mears-dfe) [inline documentation/deployments.md:20]:
```suggestion
- Pushes the image to GitHub's container registy
```

File: documentation/deployments.md [hunk 6]
```diff
@@ -8,14 +8,16 @@ This builds a Docker image from a feature branch, and then uses Terraform to cre
...
 - Builds a Docker image from code in the feature branch
 - Tags the Docker image with the tags:
     - branch name (e.g. [TEVA-1797-update-school-type-summary](https://hub.docker.com/layers/dfedigital/teaching-vacancies/TEVA-1797-update-school-type-summary/images/sha256-dc01451b1486e40a3fb1a32ca577c65ece1a28a2ff27eefbd2455202c93caa71?context=explore)). Review apps often go through several iterations, so it's worth the creation of an image in order to speed up building from cache on subsequent pushes to the branch.
     - a composite tag containing the PR number and a unique timestamp (e.g. [review-pr-2664-80dd5f417a0faabfbe3f1a4bf8570eefec07139a-20210118172840](https://hub.docker.com/layers/dfedigital/teaching-vacancies/review-pr-2664-80dd5f417a0faabfbe3f1a4bf8570eefec07139a-20210118172840/images/sha256-dc01451b1486e40a3fb1a32ca577c65ece1a28a2ff27eefbd2455202c93caa71?context=explore)). This is useful for identifying the image.
-- Logs in to Docker Hub as the service account `teachingjobs`
-- Pushes the image to Docker Hub
+- Logs in to Github's container registy as the service account `twd-tv-ci`
```
Comment (david-mears-dfe) [inline documentation/deployments.md:19]:
```suggestion
- Logs in to GitHub's container registy as the service account `twd-tv-ci`
```

File: documentation/deployments.md [hunk 7]
```diff
@@ -31,8 +33,8 @@ The GitHub actions workflow [deploy_branch.yml](/.github/workflows/deploy_branch
 
 - Builds a Docker image from code in the `dev` branch
 - Tags the Docker image with the tag
-- Logs in to Docker Hub as the service account `teachingjobs`
-- Pushes the image to Docker Hub
+- Logs in to Github's container registy as the service account `twd-tv-ci`
```
Comment (david-mears-dfe) [inline documentation/deployments.md:36]:
```suggestion
- Logs in to GitHub's container registy as the service account `twd-tv-ci`
```

File: documentation/deployments.md [hunk 8]
```diff
@@ -31,8 +33,8 @@ The GitHub actions workflow [deploy_branch.yml](/.github/workflows/deploy_branch
...
+- Pushes the image to Github packages
```
Comment (david-mears-dfe) [inline documentation/deployments.md:37]:
```suggestion
- Pushes the image to GitHub packages
```

File: documentation/deployments.md [hunk 9]
```diff
@@ -42,8 +44,8 @@ When a PR is approved and merged into `master` branch, the GitHub actions workfl
 
 - Builds and tags a Docker image from code in the `master` branch
 - Tags the Docker image with the commit SHA as the tag
-- Logs in to Docker Hub as the service account `teachingjobs`
-- Pushes the image to Docker Hub
+- Logs in to Github's container registy as the service account `twd-tv-ci`
```
Comment (david-mears-dfe) [inline documentation/deployments.md:47]:
```suggestion
- Logs in to GitHub's container registry as the service account `twd-tv-ci`
```

File: documentation/deployments.md [hunk 10]
```diff
@@ -42,8 +44,8 @@ When a PR is approved and merged into `master` branch, the GitHub actions workfl
...
+- Pushes the image to Github packages
```
Comment (david-mears-dfe) [inline documentation/deployments.md:48]:
```suggestion
- Pushes the image to GitHub packages
```

File: documentation/deployments.md [hunk 11]
```diff
@@ -97,7 +99,7 @@ Click "Run workflow", and choose:
 
 If you need to deploy a particular image to an environment, including in a rollback situation, this is possible with the [deploy_app.yml](/.github/workflows/deploy_app.yml) workflow.
 
-You will need to know the tag of the Docker image you wish to use. Go to this [Docker Hub URL with tags ordered by last_updated](https://hub.docker.com/r/dfedigital/teaching-vacancies/tags?page=1&ordering=last_updated)
+You will need to know the tag of the Docker image you wish to use. Go to this [Github packages](https://github.com/DFE-Digital/teaching-vacancies/pkgs/container/teaching-vacancies)
```
Comment (david-mears-dfe) [inline documentation/deployments.md:102]:
```suggestion
You will need to know the tag of the Docker image you wish to use. Go to the [Github packages](https://github.com/DFE-Digital/teaching-vacancies/pkgs/container/teaching-vacancies)
```

File: documentation/deployments.md [hunk 12]
```diff
@@ -123,7 +125,7 @@ performs these steps:
 
 Requirements:
 - [terraform CLI](https://www.terraform.io/downloads.html) of at least version `0.15.5`
-- Log in to Docker Hub (with `docker login`)
+- Log in to Github Packages (with `docker login ghcr.io -u USERNAME` - use PAT token as passoword)
```
Comment (david-mears-dfe) [inline documentation/deployments.md:128]:
```suggestion
- Log in to GitHub Packages (with `docker login ghcr.io -u USERNAME` - use PAT token as passoword)
```

File: documentation/docker.md [hunk 13]
```diff
@@ -226,41 +226,31 @@ For images built off the `master` branch, we use the [SHA of the GitHub commit](
 
 - [Multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) were introduced in Docker 17.05
 
-## Docker Hub
+## Github Packages (Container registry)
```
Comment (david-mears-dfe) [inline documentation/docker.md:229]:
```suggestion
## GitHub Packages (Container registry)
```

File: documentation/docker.md [hunk 14]
```diff
@@ -226,41 +226,31 @@ For images built off the `master` branch, we use the [SHA of the GitHub commit](
...
 
-### Docker Hub organisation
+### Github container registry
```
Comment (david-mears-dfe) [inline documentation/docker.md:231]:
```suggestion
### GitHub container registry
```

File: documentation/docker.md [hunk 15]
```diff
@@ -226,41 +226,31 @@ For images built off the `master` branch, we use the [SHA of the GitHub commit](
...
 
-The Docker images for Teaching Vacancies are stored in the Docker Hub organisation [dfedigital](https://hub.docker.com/u/dfedigital).
+The Docker images for Teaching Vacancies are stored in the Github's container registry [https://github.com/DFE-Digital/teaching-vacancies/](https://github.com/orgs/DFE-Digital/packages).
```
Comment (david-mears-dfe) [inline documentation/docker.md:233]:
```suggestion
The Docker images for Teaching Vacancies are stored in GitHub's container registry [https://github.com/DFE-Digital/teaching-vacancies/](https://github.com/orgs/DFE-Digital/packages).
```

Comment (david-mears-dfe) [reply documentation/docker.md:233]: Something wrong with the url: [this] doesn't match (this).

File: documentation/docker.md [hunk 16]
```diff
@@ -226,41 +226,31 @@ For images built off the `master` branch, we use the [SHA of the GitHub commit](
...
-Therefore we need to log in with Docker Team accounts.
 ### Docker Hub users
 
 #### Individual end-user
 
-To use `docker` commands, either directly, or through the abstraction of the Makefile, you must first be logged in as a Docker user.
+To use `docker` commands, either directly, or through the abstraction of the Makefile, you must first be logged in as a Github user.
 
-- You will require write access to Docker Hub [dfedigital/teaching-vacancies](https://hub.docker.com/r/dfedigital/teaching-vacancies) repository. Ask in #digital-tools-support should you require it.
+- You will require write access to Github packages [https://github.com/DFE-Digital/teaching-vacancies/](https://github.com/DFE-Digital/teaching-vacancies/pkgs/container/teaching-vacancies) repository. Ask in #digital-tools-support should you require it.
```
Comment (david-mears-dfe) [inline documentation/docker.md:242]:
```suggestion
- You will require write access to GitHub packages [https://github.com/DFE-Digital/teaching-vacancies/](https://github.com/DFE-Digital/teaching-vacancies/pkgs/container/teaching-vacancies) repository. Ask in #digital-tools-support should you require it.
```

File: documentation/docker.md [hunk 17]
```diff
@@ -226,41 +226,31 @@ For images built off the `master` branch, we use the [SHA of the GitHub commit](
...
-> Anonymous and Free Docker Hub users are limited to 100 and 200 container image pull requests per six hours.
-> Docker Pro and Docker Team accounts continue to have unlimited access to pull container images from Docker Hub.
-
-Therefore we need to log in with Docker Team accounts.
 ### Docker Hub users
 
 #### Individual end-user
 
-To use `docker` commands, either directly, or through the abstraction of the Makefile, you must first be logged in as a Docker user.
+To use `docker` commands, either directly, or through the abstraction of the Makefile, you must first be logged in as a Github user.
```
Comment (david-mears-dfe) [inline documentation/docker.md:240]:
```suggestion
To use `docker` commands, either directly, or through the abstraction of the Makefile, you must first be logged in as a GitHub user.
```

File: documentation/docker.md [hunk 18]
```diff
@@ -270,27 +260,30 @@ These are used within the Gov.UK PaaS web app and worker app [terraform](../terr
...
       with:
-        username: ${{ secrets.DOCKER_USERNAME }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GIT_HUB_SERVICE_ACCOUNT_TOKEN }}
 ```
 The secrets can not be decrypted through the UI, but can be updated.
 
-### Docker Hub repositories
+### Github container registry
```
Comment (david-mears-dfe) [inline documentation/docker.md:280]:
```suggestion
### GitHub container registry
```

File: documentation/docker.md [hunk 19]
```diff
@@ -270,27 +260,30 @@ These are used within the Gov.UK PaaS web app and worker app [terraform](../terr
...
+
+> Docker images are pushed to Github container registry through the `docker push` command. A single Github container registry can hold many Docker images (stored as *tags*).
```
Comment (david-mears-dfe) [inline documentation/docker.md:282]:
```suggestion
Docker images are pushed to the GitHub container registry through the `docker push` command. A single GitHub container registry can hold many Docker images (stored as *tags*).
```

Is this paragraph a quote?

File: documentation/docker.md [hunk 20]
```diff
@@ -270,27 +260,30 @@ These are used within the Gov.UK PaaS web app and worker app [terraform](../terr
...
+
+To view the docker images stored on Github's container registry, you need to go through DfE's main Github page, and then click packages. Please note, Github repositories are different from Github packages:
+- [Teaching-vacancies Packages (docker images)](https://github.com/DFE-Digital/teaching-vacancies/pkgs/container/teaching-vacancies)
 
-Note that [Docker Hub repositories](https://docs.docker.com/docker-hub/repos/) are distinct from GitHub repositories:
-> Docker Hub repositories allow you to share container images with your team, customers, or the Docker community at large.
-> Docker images are pushed to Docker Hub through the `docker push` command. A single Docker Hub repository can hold many Docker images (stored as *tags*).
+### Docker image scam
```
Comment (david-mears-dfe) [inline documentation/docker.md:287]:
```suggestion
### Docker image scan
```

File: documentation/docker.md [hunk 21]
```diff
@@ -270,27 +260,30 @@ These are used within the Gov.UK PaaS web app and worker app [terraform](../terr
...
 
-Repositories include the organisation name, but in the URL, note that the `u` for user, changes to `r` for repository:
-- [https://hub.docker.com/r/dfedigital/teaching-vacancies](https://hub.docker.com/r/dfedigital/teaching-vacancies)
+As part of the CI\CD, we conduct docker security scan – using anchore/scan-action@v2, which uses Anchore Engine. This allowing a deep image inspection and vulnerability scanning, which allows detailed analysis of container workloads, producing reports and defining policies that can be used in the software delivery lifecycle stack. The stateless scan result (SARIF format) is loaded to Github’s Security scanning alert console using github/codeql-action/upload-sarif@v1
```
Comment (david-mears-dfe) [inline documentation/docker.md:289]:
```suggestion
As part of the CI/CD, we conduct a Docker security scan using anchore/scan-action@v2, which uses Anchore Engine. This allows a deep image inspection and vulnerability scan, which allows detailed analysis of container workloads, producing reports and defining policies that can be used in the software delivery lifecycle stack. The stateless scan result (SARIF format) is loaded to Github’s Security scanning alert console using github/codeql-action/upload-sarif@v1
```

**Review by alexbowen — APPROVED** [review]
makes sense to me

**Pull request merged**